### DISCUSSION TITLE
Detailed CSV Export in V2

### DISCUSSION THREAD

**Discussion by VoxilienVoxil** [danniehansen/activity-timer#59] [Ideas]
Hi Dannie, we use activity timer for a while now. With V2, we subscribed to pro plan because we regularly export CSV from "Activity timer history board button". But one thing I miss is "old" detailed CSV export. Current implementation doesn't export "Start date and time","End date and time" columns. Is it possible to add them to "new history board button". Or may be adding a second export option that outputs all columns? (like the old one)

**Comment by danniehansen** [comment]
Hi @VoxilienVoxil,

First of all. Thank you so much for supporting my journey with Activity timer by choosing to go pro. Very much appreciated.

I'm actually sitting right now extending the data exporter with extra filtering options. I'll add it to the list & get this done as well.
To to be sure that I understand the request correctly. You would like to have a option to export all of the tracking's separately correct? So if I made 5 separate tracking's on a card they would appear separately?

Also - if you got any other requests on useful features that you think could make for a great addition. Don't hesitate to shoot them my way!

**Comment by danniehansen** [comment]
@VoxilienVoxil I think I got a pretty good idea of what you was thinking. I've implemented 3 variations on the data using a 'Group by' feature. By default it will show all tracking's separately 1 per row. You can then choose to group the trackings by card og by card + member.

In addition it also now displays the start datetime / end datetime. When not using grouping this is the from/to of a specific tracking session.

Also introduced a new 'Data exporter - Estimates' feature. Scrum masters will love this!

Let me know your thoughts on this. I'll gladly take feedback.

**Reply by VoxilienVoxil** [reply]
Hi @danniehansen, First of all, thank you for implementing a much more comprehensive & flexible filtering so quickly. Very much appreciated. 

I didn't get the chance to test thoroughly but it seems like when you apply some column filters, same row added to output table more than once. Also if you set the "Date from" and the "Date to" same day, query returns other dates as well. I will report it properly as an issue when I get the chance.

Thanks again.

**Reply by danniehansen** [reply]
Hi @VoxilienVoxil,

No problem at all. Thank you for reporting this issue - I'm looking into it right away.

Regarding showing the same row more than once. This is most likely due to grouping. Default is no grouping (also a filter). There are 3 grouping variants:

1. No grouping = If 2 separate users makes 3 trackings on a card. Then there would be 6 different rows for that card. 1 tracking = 1 row. This makes it easier to import into external system if you need down to the minute accuracy.

2. Card = Here we have only 1 row per card. Member ids & member names are comma separated, start / end datetime reflect the time of the first tracking & the time of the last tracking and last but not least - time spent is all added together cross members & different tracking, just the content sum.

3. Card & member = Here we have 1 rows per card, per member on the card. So if 2 users each made 3 trackings. Then it'll have 2 rows, one for each member.

These 3 variants allow for various data exports. I'm still trying to figure out which option is best to leave as the default option. I'm actually thinking of switching it over to 'Card & member' so that by default 1 row = 1 card.

Let me know your thoughts on this. Still looking for input :) Will get back to you as soon as possible with a fix when you combine both date start & end on the filtering.

**Reply by danniehansen** [reply]
@VoxilienVoxil Found what you was mentioning. Tried the filtering out on a large data set & there I found that time trackings duplicated in the UI when filtering. This should be solved now & this also solved an issue in edge where header columns kept displaying even when removed.
